File: Assignment/Abstraction/Expressions/BooleanExpressionNode.cs
using Assignment.Extensions;
using System;

namespace Assignment.Abstraction.Expressions
{
    internal struct BooleanExpressionNode : IExpressionNode
    {
        public BooleanExpressionNode(INode left, INode right, BooleanOperations operation)
        {
            this.Operation = operation;
            this.Left = left;
            this.Right = right;
        }

        public BooleanOperations Operation { get; }
        public INode Left { get; }
        public INode Right { get; }

        public string DisplayName()
        {
            return $"{nameof(BooleanExpressionNode)}(operation: {this.Operation.GetDisplayName()})";
        }
    }
}

File: src/main/java/compiler/abstraction/expressions/BooleanExpressionNode.java
package compiler.abstraction.expressions;

import java.util.UUID;

import compiler.abstraction.BooleanOperation;
import compiler.abstraction.INode;

public class BooleanExpressionNode extends INode {
    public final INode left;
    public final INode right;
    public final BooleanOperation operation;

    public BooleanExpressionNode(INode left, INode right, BooleanOperation operation) {
        super();
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    public BooleanExpressionNode(INode left, INode right, BooleanOperation operation, UUID id) {
        super(id);
        this.left = left;
        this.right = right;
        this.operation = operation;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(operation: %s)", BooleanExpressionNode.class.getSimpleName(), this.operation);
    }
}